### PR TITLE
Fix usage of uninitialized memory in swift_addNewDSOImage.

### PR DESCRIPTION
**Pull request by pschuh**
This has an upstream PR: https://github.com/apple/swift/pull/26275, but that hasn't merged yet. Trying to get it in tensorflow branch sooner to support leak-checking.

### CODE REVIEW

**Comment by pschuh** [comment]
@swift-ci please test

**Comment by pschuh** [comment]
@swift-ci please clean test tensorflow

File: stdlib/public/runtime/ImageInspectionMachO.cpp [hunk 1]
```diff
@@ -133,20 +133,19 @@ void addImageCallback2Sections(const mach_header *mh, intptr_t vmaddr_slide) {
...
-                     addImageProtocolsBlockCallback>);
+  REGISTER_FUNC(addImageCallback<TextSegment, ProtocolsSection,
+                                 addImageProtocolsBlockCallbackUnsafe>);
 }
 
 void swift::initializeProtocolConformanceLookup() {
   REGISTER_FUNC(
-    addImageCallback<TextSegment, ProtocolConformancesSection,
-                     addImageProtocolConformanceBlockCallback>);
+      addImageCallback<TextSegment, ProtocolConformancesSection,
```
Comment (bgogul) [inline stdlib/public/runtime/ImageInspectionMachO.cpp]: Indentation seems off here and below?

**Review by bgogul — APPROVED** [review]

File: stdlib/public/runtime/ImageInspectionMachO.cpp (hunk 1 above)
Comment (pschuh) [reply stdlib/public/runtime/ImageInspectionMachO.cpp]: This is clang-formatted.

**Pull request merged**